Add command for migrating templates from rpmdb

marmarek opened the pull request:
Fixes QubesOS/qubes-issues#7436

While at it, fix few other template related things:
- https://github.com/QubesOS/qubes-issues/issues/7426
- https://github.com/QubesOS/qubes-issues/issues/7094

marmarek committed:
Add command for migrating templates from rpmdb

Fixes QubesOS/qubes-issues#7436

DemiMarie commented on qubesadmin/tools/qvm_template.py:
```diff
@@ -1574,6 +1578,50 @@ def repolist(args: argparse.Namespace, app: qubesadmin.app.QubesBase) -> None:
+            continue
+        pkgs_to_remove.append(pkg)
+    subprocess.check_call(
+        ['rpm', '-e', '--justdb'] + [p['name'] for p in pkgs_to_remove])
```
Can you use the librpm Python bindings instead of a subprocess here?

marmarek replied:
I'd prefer not to, to not screw it up.

DemiMarie reviewed (approved):
Looks good, though there is a trivial optimization opportunity.

marmarek committed:
ci: switch unit tests back to docker

marmarek committed:
pylint: disable consider-using-f-string for now

marmarek committed:
make pylint happy

adjust for:
 - unspecified-encoding
 - use-list-literal
 - redundant-u-string-prefix

marmarek committed:
qvm-template: Clarify tripple confirmation on 'purge' action

Fixes QubesOS/qubes-issues#7094

marmarek committed:
qvm-template-postprocess: fix installing templates with small rootfs

There is already support for importing volume of a specific size, no
need for calling resize manually. This fixes template install on storage
pools that do not support resize (shrinking).

Fixes QubesOS/qubes-issues#7426

marmarek commented:
Manual test passed.

DemiMarie commented on qubesadmin/tests/tools/qvm_template.py:
```diff
@@ -5414,3 +5417,105 @@ def test_300_repo_files_glob(self, mock_repolist):
+            'qubes-template-test-existing',
+            'qubes-template-test-migrated',
+            'qubes-template-test-removed'])
+        self.assertAllCalled()
```
nit: missing newline at EOF

DemiMarie commented on qubesadmin/tools/qvm_template.py:
```diff
@@ -1574,6 +1580,56 @@ def repolist(args: argparse.Namespace, app: qubesadmin.app.QubesBase) -> None:
+            continue
+        pkgs_to_remove.append(pkg)
+    subprocess.check_call(
+        ['rpm', '-e', '--justdb'] +
```
```suggestion
        ['rpm', '-e', '--justdb', '--'] +
```

in case the package name starts with `-` (it shouldn’t, but better to be safe)

marmarek replied:
That's impossible - the loop above explicitly collects only packages with `qubes-template-` prefix.

DemiMarie replied:
That wasn’t obvious from looking at the code.  Would it be possible to collect a list of strings instead of a list of package objects, and add an assertion here?

marmarek commented:
PipelineRefresh

DemiMarie commented on qubesadmin/tools/qvm_template.py:
```diff
@@ -1444,9 +1444,11 @@ def remove(
         # XXX: Mutating the list later seems to break the tests...
         remove_list_copy = remove_list.copy()
-        for _ in range(repeat):
+        for confirm_n in range(repeat):
```
Nit: typo in commit message (`tripple` should be `triple`)
